include/libopencm3/ethernet/mac_stm32fxx7.h

```diff
@@ -247,45 +259,48 @@
 /*---------------------------------------------------------------------------*/
 /* MACDBGR -------------------------------------------------------------------*/
 
-#define ETH_MACDBGR_MMRPEA		(1<<0)
-#define ETH_MACDBGR_MSFRWCS		(3<<1)
-#define ETH_MACDBGR_RFWRA		(1<<4)
-
-#define ETH_MACDBGR_RFRCS_SHIFT		5
-#define ETH_MACDBGR_RFRCS		(3<<ETH_MACDBGR_RFRCS_SHIFT)
-#define ETH_MACDBGR_RFRCS_IDLE		(0<<ETH_MACDBGR_RFRCS_SHIFT)
-#define ETH_MACDBGR_RFRCS_RDATA		(1<<ETH_MACDBGR_RFRCS_SHIFT)
-#define ETH_MACDBGR_RFRCS_RSTAT		(2<<ETH_MACDBGR_RFRCS_SHIFT)
-#define ETH_MACDBGR_RFRCS_FLUSH		(3<<ETH_MACDBGR_RFRCS_SHIFT)
-
-#define ETH_MACDBGR_RFFL_SHIFT		8
-#define ETH_MACDBGR_RFFL		(3<<ETH_MACDBGR_RFFL_SHIFT)
-#define ETH_MACDBGR_RFFL_EMPTY		(0<<ETH_MACDBGR_RFFL_SHIFT)
-#define ETH_MACDBGR_RFFL_BELOW		(1<<ETH_MACDBGR_RFFL_SHIFT)
-#define ETH_MACDBGR_RFFL_ABOVE		(2<<ETH_MACDBGR_RFFL_SHIFT)
-#define ETH_MACDBGR_RFFL_FULL		(3<<ETH_MACDBGR_RFFL_SHIFT)
-
-#define ETH_MACDBGR_MMTEA		(1<<16)
-
-#define ETH_MACDBGR_MTFCS_SHIFT		17
-#define ETH_MACDBGR_MTFCS		(3 << ETH_MACDBGR_MTFCS_SHIFT)
-#define ETH_MACDBGR_MTFCS_IDLE		(0 << ETH_MACDBGR_MTFCS_SHIFT)
-#define ETH_MACDBGR_MTFCS_WAIT		(1 << ETH_MACDBGR_MTFCS_SHIFT)
-#define ETH_MACDBGR_MTFCS_PAUSE		(2 << ETH_MACDBGR_MTFCS_SHIFT)
-#define ETH_MACDBGR_MTFCS_TRANSFER	(3 << ETH_MACDBGR_MTFCS_SHIFT)
-
-#define ETH_MACDBGR_MTP			(1<<19)
-
-#define ETH_MACDBGR_TFRS_SHIFT		20
-#define ETH_MACDBGR_TFRS		(3<<ETH_MACDBGR_TFRS_SHIFT)
-#define ETH_MACDBGR_TFRS_IDLE		(0<<ETH_MACDBGR_TFRS_SHIFT)
-#define ETH_MACDBGR_TFRS_READ		(1<<ETH_MACDBGR_TFRS_SHIFT)
-#define ETH_MACDBGR_TFRS_WAIT		(2<<ETH_MACDBGR_TFRS_SHIFT)
-#define ETH_MACDBGR_TFRS_FLUSH		(3<<ETH_MACDBGR_TFRS_SHIFT)
-
-#define ETH_MACDBGR_TFWA		(1<<22)
-#define ETH_MACDBGR_TFNE		(1<<24)
-#define ETH_MACDBGR_TFF			(1<<25)
+/* Following registers not available on F1 */
+#if !defined(STM32F1)
```

Comment:
if you switch this to checking on the definition of hte register, it will automatically work if/when more parts with /without the registers appear.  If this is actually a lot, we might want to add it as an extra header